### ISSUE TITLE
macOS: detect high contrast theme

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#105481]
With the Electron 9 update came support to detect high-contrast or inverted-color themes on macOS. We currently only detect this on Windows and can now enable this for macOS as well.

//cc @isidorn @aeschli @deepak1556

**Comment by isidorn** [comment]
Great. I think this makes perfect sense and we should go with this. If users complain we can easily change it.

**Comment by bpasero** [comment]
**Verification:** 
* make sure on macOS you see the `window.autoDetectHighContrast` setting
* verify that we change to the user selected HC theme when changing to high contrast on macOS via settings -> accessibility -> display (either "Invert colors" or "Increase contrast")
* verify this change happens while the app is running
* verify this change happens when you start the app and have not previously selected a HC theme while being in HC mode in the OS

![image](https://user-images.githubusercontent.com/900690/91418612-addb1580-e852-11ea-9821-2f87cfa8ded9.png)

**Comment by kylehalleman** [comment]
Just a note, this was very jarring when updating to the latest version. I had to come to this issue to find out it was something I could override by going to user settings for color theme. I would suggest that this auto-selecting of the high contrast theme only take effect if the user is using the default theme. I had already customized a theme that suited me very well and to have it replaced because I have "Increase contrast" accessibility setting on was unexpected.
